lib/ai/embeddingprocessor_test.go

```diff
@@ -123,6 +123,28 @@ func TestNodeEmbeddingGeneration(t *testing.T) {
 	validateEmbeddings(t,
 		presence.GetNodeStream(ctx, defaults.Namespace),
 		embeddings.GetEmbeddings(ctx, types.KindNode))
+
+	for k, v := range embedder.timesCalled {
+		require.Equal(t, 1, v, "expected %v to be computed once, was %d", k, v)
+	}
+
+	// Run once more and verify that only changed nodes get their embeddings recalculated
+	node1 := nodes[0]
+	node1.GetMetadata().Labels["foo"] = "bar"
+	_, err = presence.UpsertNode(ctx, node1)
+	require.NoError(t, err)
+
+	ctx, cancel = context.WithCancel(context.Background())
+	defer cancel()
+	processor.Process(ctx)
```

Comment:
Is Process async? Why do we need the cancelable context here?

Reply:
Good point, we do not!